bot/TS3Bot.py

```diff
@@ -331,7 +331,7 @@ def setResetroster(self, date, red=[], green=[], blue=[], ebg=[]):
     def getActiveCommanders(self):
         return self.commander_checker.execute()
 
-    def removeGuild(self, name):
+    def removeGuild(self, name, tag=None):
```

Comment:
why is tag an optional parameter here? Should we not be able to determine the tag based on the name from the API in any case?

Reply:
True, but Channels can/will be created with custom Tags.
The created channel thereby does not have the "official" Guild tag, and can not be found for deletion.

Reply:
The name of the channels should always use the actual ingame-tag. The custom tags are just meant to deal with duplicate guild-groups in TS and should strictly be used as such. They can be looked up by the guild name in the `guilds`-table of the db:

```sql
SELECT ts_group FROM guilds WHERE guild_name = 'Unexpected Chaos'
```
Should yield `uC.`.

Reply:
The tag provided on the create command is uses for the channel name: bot/TS3Bot.py:424
On delete the tag will be fetched from the api, this prevents deletion of the channel if a custom tag was used on create.

Reply:
That is working as intended. The custom tag you are referring to is the parameter `groupname` of the same function. That one is indeed optional, as it is the same as `tag` in any case where the tag of a guild can serve as a unique group name. The parameter `tag` should always hold the actual tag of the guild. It is only a parameter because no lookup on the API is done so far, but that is an outstanding issue https://github.com/FlussuferOrga/ts-gw2-verifyBot/issues/32
We could very well look up the tag from the API and drop the `tag`-parameter alltogether from `createGuild`.

Reply:
groupname and tag can be different.
```
POST /guild
{
  "name": "Stardust Alliance",
  "tag": "Durst",
  "groupname": "DUST,
  "contacts": [
    "UserName.1234"
  ]
}
```
Will create the channel `Stardust Alliance [Durst]`

```
DELETE /guild
{
  "name": "Stardust Alliance"
}
```
Does **not** delete the channel.

Reply:
So dropping the Tag parameter on create and delete is the way to go?

Reply:
That is correct, there are no checks in place to ensure the tag that is being passed is the actual tag used in game. The intended use is for the caller to make sure name and tag are correct and to provide an optional groupname in case a group of that name already exists. This is by no means optimal, especially since deleting a guild causes issues if someone accidentally created a guildchannel for a guild that does not exist, which is checked upon deletion, but not during creation. That simply happend due to the feature being rushed. Dropping the tag parameter in both functions and looking it up from the API in both cases as you suggested is indeed the way to go. That would be a possible solution for #32 .